### DISCUSSION TITLE
Combobox with custom action

### DISCUSSION THREAD

**Discussion by Cuttsy27** [adobe/react-spectrum#5042] [Q&A]
Hi,

I'm wondering if it is possible to add a custom action item to the ComboBox and other similar components like Select. The action/s would sit at the bottom of the list, not be filtered with other items, but still be navigated to with arrow keys like the other items. The action could be used for things like "Add more" which might open a modal to add an item to the list.

Is there a preferred way to do this for accessibility or is it simply adding it as a normal item then checking for it onSelectionChange and onInputChange (to stop it from being filtered)?

**Comment by reidbarber** [comment]
I think the way you described is your best bet. Use controlled selection where `onSelectionChange`, you can trigger the action if that item is selected, but not update make that item selected in the collection. And use custom filtering to always show that item last.

Somewhat related, but there is a WIP PR to [Support links in collection components](https://github.com/adobe/react-spectrum/pull/4993#top).
